### PR TITLE
ISSUE-949 [ICS Reply] ICS patching core process

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/949

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[ERROR] Failed to execute goal io.github.evis:scalafix-maven-plugin_2.13:0.1.6_0.9.34:scalafix (scala-check-style) on project jmap-extensions: Scalafix invoked with errors. Check logs for details. -> [Help 1]
```

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/CalendarEventReplyGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          .stream()
+          .filter(!_.getName.equals("ATTENDEE"))
+          .forEach(e => propertyList.add(e))
+        propertyList.add(buildReplyAttendee(attendee, partStat))
+        propertyList
+      }).get
+
+    val calendar = new Calendar()
+      .withDefaults()
+      .withComponent(new VEvent(replyProperty).getFluentTarget)
```
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/CalendarEventReplyGenerator.scala:31]: Don't forget to copy the `VTIMEZONE` component too :)

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/CalendarEventReplyGenerator.scala [hunk 2]
```diff
@@ -3,42 +3,56 @@ package com.linagora.tmail.james.jmap.model
...
           .forEach(e => propertyList.add(e))
-        propertyList.add(buildReplyAttendee(attendee, partStat))
+        propertyList.add(buildReplyAttendee(attendeeReply))
         propertyList
       }).get
 
     val calendar = new Calendar()
       .withDefaults()
       .withComponent(new VEvent(replyProperty).getFluentTarget)
+      .withProdId(PRODID)
```
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/CalendarEventReplyGenerator.scala:41]: Is the prodid necessary? TMail calendar does not really exist as well right?^^'

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventParsedTest.scala [hunk 3]
```diff
@@ -477,6 +482,83 @@ class CalendarEventParsedTest {
     }
   }
 
+  @Test
+  def test1(): Unit = {
```
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventParsedTest.scala:486]: Need a proper test name

**Review by Arsnael — APPROVED** [review]

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/CalendarEventReplyGenerator.scala (hunk 2 above)
Comment (quantranhong1999) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/CalendarEventReplyGenerator.scala:41]: We likely need to override the sender PROID though (origin e.g. Google Calendar). Not sure if we can let it empty though.

Maybe keep Sabre or somethign currently as OpenPaaS calendar?

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+          | OMPTETEST15.LINAGORA@DOMAIN.TLD:DTSTAMP:20240222T204008Z
+          |SEQUENCE:0
+          |ATTENDEE;ROLE=REQ-PARTICIPANT;PARTSTAT=DECLINED;CUTYPE=INDIVIDUAL:mailto:bob@domain.com
+          |END:VEVENT
+          |END:VCALENDAR
+          |""".stripMargin)
+  }
+
+  @Test
+  def shouldThrowWhenMethodIsNotRequest(): Unit = {
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:110]: I think NO
The REPLY should be strict based on REQUEST
No Request, No Reply

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+          |UID:8eae5147-f2df-4853-8fe0-c88678bc8b9f
+          |TRANSP:OPAQUE
+          |DTSTART;TZID=Europe/Paris:20240223T160000
+          |DTEND;TZID=Europe/Paris:20240223T163000
+          |CLASS:PUBLIC
+          |SUMMARY:Simple event
+          |ORGANIZER;CN=comptetest15.linagora@domain.tld:mailto:comptetest15.linagora@a
+          | EXAMPLE.COM:ATTENDEE\;PARTSTAT=NEEDS-ACTION\;RSVP=TRUE\;ROLE=REQ-PARTICIPANT\;CUTYPE=INDIVI
+          | DUAL:mailto:btellier@example.com
+          | OMPTETEST15.LINAGORA@DOMAIN.TLD:DTSTAMP:20240222T204008Z
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:101]: ah, I got it, the line separator issue

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+object CalendarEventReplyGeneratorTest {
+  val partStats: Stream[Arguments] = {
+    Stream.of(
+      Arguments.of(PartStat.ACCEPTED),
+      Arguments.of(PartStat.DECLINED),
+      Arguments.of(PartStat.TENTATIVE))
+  }
+}
+
+class CalendarEventReplyGeneratorTest {
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:29]: Edge case idea: what happen when I'm not attending the event (not listed as an attendee)?

Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:29]: I just tested on GMAIL
A send invite to B
B forwarded invite email to C
C click YES
-> A received accepted email from C

Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:29]: Checked the `ics` file
The request ics file: NO ATTENDEE C
The reply ics file: Remove B, and replace by C

Comment (Arsnael) [reply tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:29]: I don't think this behavior is correct then? If C was not invited initially, he should not be able to do an action on it even if he got it forwarded by the initial attendee?

Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:29]: I think so, the view from the user case is incorrect.

Comment (quantranhong1999) [reply tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:29]: Argee on that too.

Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventReplyGeneratorTest.scala:29]: updated

**Review by Arsnael — APPROVED** [review]
